### PR TITLE
Add dosage extension for non-decimal dose strings

### PR DESCRIPTION
**Pull request by kivarsson-va**
Added/updated some tests and add a couple blank checks to the work Bryan did with this.

After looking at the CDW query results from KBS, we opted to leave out all parsing from the API since any parsing we need for our app is going to be pretty implementation-specific and probably isn't best for a broader API. Everything will just go into the originalText extension.

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java [hunk 1]
```diff
@@ -27,33 +28,46 @@
 
   /** Create a dosage from a sig and ptInstruction. */
   public static Dosage dosageInstruction(String sig, String ptInstructions, List<Dose> doses) {
+    List<DoseAndRate> doseAndRates = null;
 
-    if (allBlank(sig, ptInstructions, doses)) {
-      return null;
+    if (!isBlank(doses)) {
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:33]: I think this is a bit over-complicated. `Safe.stream()` will create an empty stream if `doses` is `null`, ultimately ending with creating an empty list. The `allBlank` method on line 42 will check for null, as well as empty lists. 

Instead, you could set doseAndRates using Safe.stream, and use `emptyToNull(doseAndRates)` on line 49 to trim the list to null if it's empty.

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformersTest.java [hunk 2]
```diff
@@ -20,14 +22,43 @@
...
 class R4MedicationTransformersTest {
-
   private final R4MedicationTransformers _tx =
       Mockito.mock(R4MedicationTransformers.class, Mockito.CALLS_REAL_METHODS);
 
   static CodeableConcept _cc(String text, List<Coding> codings) {
     return CodeableConcept.builder().text(text).coding(codings).build();
   }
 
+  static Dosage _dosage(
```
Comment (jhulbert-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformersTest.java:32]: I think this should be two methods. It's creating a sample, so I don't think it should do complex logic. What about `_dosageWithExtension` and `_dosageWithDoseAndRate`?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java (hunk 1 above)
Comment (kivarsson-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:33]: okay yeah thanks, we kept refactoring this to try to make it as uncomplicated as possible and without unnecessary data structures but i like your suggestion as a very straightforward way. forgot about the `emptyToNull` helper, too 🤦‍♀️

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformersTest.java (hunk 2 above)
Comment (kivarsson-va) [reply vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformersTest.java:32]: agreed, will do

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**